src/main/java/randoop/condition/OperationConditions.java

```diff
@@ -0,0 +1,174 @@
+package randoop.condition;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * The collection of all pre-, return-, and throws-conditions defined on this operation and
+ * inherited from supertypes.
+ */
+public class OperationConditions {
+
+  /** The pre-conditions for the operation */
+  private final List<Condition> preconditions;
+
+  /** The return-conditions. */
+  private final List<PrePostConditionPair> prePostConditionPairs;
+
+  /** The throws-conditions. */
+  private final List<PreThrowsConditionPair> preThrowsConditionPairs;
+
+  /**
+   * Mirrors the overrides/implements relation among methods. If this OperationConditions is the
+   * local specification for method m, the {@code parentList} contains one element for each method
+   * that m overrides or implements.
+   *
+   * <p>For an operation that is a method, the {@link OperationConditions} form an arbitrary
+   * directed acyclic graph consisting of conditions for methods of supertypes, each of which has
+   * attached specifications.
+   */
+  private List<OperationConditions> parentList = new ArrayList<>();
+
+  /** Creates an empty {@link OperationConditions} object. */
+  OperationConditions() {
+    this(
+        new ArrayList<Condition>(),
+        new ArrayList<PrePostConditionPair>(),
+        new ArrayList<PreThrowsConditionPair>());
+  }
+
+  /**
+   * Creates an {@link OperationConditions} object for the given pre-conditions, pre-post-condition
+   * pairs, and pre-throws condition pairs.
+   *
+   * @param preconditions the pre-conditions
+   * @param prePostConditionPairs the pre-post-condition pairs
+   * @param preThrowsConditionPairs the pre-throws condition pairs
+   */
+  OperationConditions(
+      List<Condition> preconditions,
+      List<PrePostConditionPair> prePostConditionPairs,
+      List<PreThrowsConditionPair> preThrowsConditionPairs) {
+    this.preconditions = preconditions;
+    this.prePostConditionPairs = prePostConditionPairs;
+    this.preThrowsConditionPairs = preThrowsConditionPairs;
+  }
+
+  /**
+   * Check the pre-conditions for this operation against the arguments. Constructs an {@link
+   * ExpectedOutcomeTable} for this operation, and for this operation in all supertypes.
+   *
+   * @param args the argument values to test the preconditions
+   * @return the table with entries for this operation
+   * @see #check(Object[], ExpectedOutcomeTable)
+   */
+  public ExpectedOutcomeTable check(Object[] args) {
```

Comment:
This doesn't check the entire OperationConditions, so I think it should be named `checkPrestate` instead.  You could make the same naming change to other methods in this class that start with `check`.